src/ui/components/FocusContextReduxAdapter.tsx

```diff
@@ -1,37 +1,33 @@
 import { TimeStampedPointRange } from "@replayio/protocol";
-import { PropsWithChildren, useCallback, useDeferredValue, useMemo } from "react";
+import { PropsWithChildren, useCallback, useContext, useDeferredValue, useMemo } from "react";
 
 import { FocusContext, UpdateOptions } from "replay-next/src/contexts/FocusContext";
+import { SessionContext } from "replay-next/src/contexts/SessionContext";
 import { useCurrentFocusWindow } from "replay-next/src/hooks/useCurrentFocusWindow";
 import { TimeRange } from "replay-next/src/types";
-import {
-  enterFocusMode,
-  setFocusWindow as setDisplayedRange,
-  setFocusWindowImprecise as setDisplayedRangeImprecise,
-  syncFocusedRegion,
-  updateFocusWindowParam,
-} from "ui/actions/timeline";
+import { enterFocusMode, requestFocusWindow, setDisplayedFocusWindow } from "ui/actions/timeline";
 import { getFocusWindow } from "ui/reducers/timeline";
 import { useAppDispatch, useAppSelector } from "ui/setup/hooks";
 
 // Adapter that reads focus region (from Redux) and passes it to the FocusContext.
 export default function FocusContextReduxAdapter({ children }: PropsWithChildren) {
   const dispatch = useAppDispatch();
+  const { duration } = useContext(SessionContext);
   const range = useCurrentFocusWindow();
   const rangeForDisplay = useAppSelector(getFocusWindow);
 
   const rangeForSuspense = useDeferredValue(range);
   const isPending = rangeForSuspense !== range;
 
   const update = useCallback(
-    async (value: TimeStampedPointRange | null, options: UpdateOptions) => {
+    async (value: TimeStampedPointRange, options: UpdateOptions) => {
       const { sync } = options;
-
-      await dispatch(setDisplayedRange(value));
-
       if (sync) {
```

Comment:
The name of this param (`sync`) always confuses me. I know you didn't add it with this PR but since we're in here cleaning things up– I wonder if there's some other, more intuitive name we could think of? Even if we just made it longer...like "sync remote" or something.

Just thinking out loud. No action required.